### DISCUSSION TITLE
Is it possible to adapt Phogrammetrical dense point cloud to use as an input on lidR?

### DISCUSSION THREAD

**Discussion by tatopm91** [r-lidar/lidR#478] [Q&A] [answered]
I was wondering if it would be possible to use as an input for you package a dense point cloud exported from a Photogrammetrical process (x,y,z) in order to segment this forest pointcloud with the same workflow.

I tried it and besides some warnings it worked until the moment I used the function segment_trees( ) where it came the error:

"Error: No attribute 'ReturnNumber' found. This attribute is needed to extract first returns"

I understand this is a unique attribute that is contained in a pointcloud obtained by LiDAR. But it could be possible to create this attribute for my pointcloud in order to continue the workflow and detect individual trees?

**Comment by Jean-Romain** [comment, marked as answer]
Add it and fill it with ones

```r
las@data$ReturnNumber <- 1L
```

**Reply by tatopm91** [reply]
Thank you very much! It worked perfectly
